utilities/backupable/backupable_db.cc

```diff
@@ -1424,14 +1442,20 @@ IOStatus BackupEngineImpl::CreateNewBackupWithMetadata(
     item.result.wait();
     auto result = item.result.get();
     item_io_status = result.io_status;
+    Temperature temp = result.expected_src_temperature;
+    if (result.current_src_temperature != Temperature::kUnknown &&
+        (temp == Temperature::kUnknown ||
```

Comment:
Does it mean that if the DB runs with non tiered storage mode, we will always go to check temperature from the file system?

Reply:
It looks like filesystem is always queried no matter the storage mode.

Reply:
The FileSystem object is always queried, which for non-custom FS is a no-op. Therefore, putting this behind an option just creates one more way to do *the wrong thing* with probably no tangible benefit vs. just always trying to do *the right thing*. Querying the Temperature of an open file should be an in-memory operation, because the custom FileSystem should know what tier it is talking to. And if you really want to hide Temperature info or avoid querying, you can do that with a FileSystem wrapper that always returns Temperature::kUnknown.